tests/unit_tests/udf_wrapper_params/question_answering/multiple_locations_multiple_batch_complete.py

```diff
@@ -0,0 +1,59 @@
+from tests.unit_tests.udf_wrapper_params.question_answering.\
+    mock_question_answering import \
+    MockQuestionAnsweringFactory, MockQuestionAnsweringModel, MockPipeline
+
+
+def udf_wrapper():
+    from exasol_udf_mock_python.udf_context import UDFContext
+    from exasol_transformers_extension.udfs.question_answering_udf import \
+        QuestionAnswering
+    from tests.unit_tests.udf_wrapper_params.question_answering. \
+        mock_sequence_tokenizer import MockSequenceTokenizer
+    from tests.unit_tests.udf_wrapper_params.question_answering.\
+        multiple_locations_multiple_batch_complete \
+        import MultipleModelLocationsMultipleBatchComplete \
+        as params
+
+    udf = QuestionAnswering(
+        exa,
+        batch_size=params.batch_size,
+        pipeline=params.mock_pipeline,
+        base_model=params.mock_factory,
+        tokenizer=MockSequenceTokenizer)
+
+    def run(ctx: UDFContext):
+        udf.run(ctx)
+
+
+class MultipleModelLocationsMultipleBatchComplete:
```

Comment:
I think, we don't need tests for all combinations for batches and multiple models, I think, there was a misunderstanding, if we have them we can keep them.

Reply:
In the current situation, we have three different setup:
1. single model
2. multiple model
3. multiple locations (sub_dir, bfs combinations)

We should keep the third one, since we slice data based on a combination
I think, the second one can be merged with the third one.

Reply:
I would like to keep the multiple models ones, because they are less functionality. As I said, I would add only the following multiple location test, to test exactly this feature. 

1.  MultipleBucketFSConSingleSubdirSingleModelNameSingleBatch : We switch the location in a single batch
2.  MultipleBucketFSConSingleSubdirSingleModelNameMultipleBatch : We switch the location in the second batch
3.  MultipleBucketFSConSingleSubdirSingleModelNameSingleBatch : We switch the location in a single batch
4.  SingleBucketFSConMultipleSubdirSingleModelNameMultipleBatch : We switch the location in the second batch

Reply:
we filter the data based on this function:
 `   
model_df = batch_df[
                (batch_df['model_name'] == model_name) &
                (batch_df['bucketfs_conn'] == bucketfs_conn) &
                (batch_df['sub_dir'] == sub_dir)]`
Is it necessary to have separated tests for each item of this combination to ensure that this filtering is working correctly?

I suggested combining MultipleLocations and MultipleModels tests, as I thought it would be sufficient to test different combinations in different batches.

Reply:
Tests should not assume any specific implementation and they should test as good as possible orthogonal features. Here, are the orthogonal features "handling multiple models with different batching" and "handling different locations"